scripts/dev_setup.sh

```diff
@@ -14,7 +14,7 @@
 # 3 ${HOME}/bin/, or ${INSTALL_DIR} is expected to be on the path - hashicorp tools/etc.  will be installed there on linux systems.
 
 # fast fail.
-set -eox pipefail
```

Comment:
I think we should keep `-x` so we can see what's going on in the script

Reply:
That seems useful for debugging, but it's really too noisy for everyday use.  I note it was added without discussion only recently (May 17).

Reply:
I'm tempted to wrap every command with side-effects with (set -x; ...) as discussed [here](https://stackoverflow.com/questions/2853803/how-to-echo-shell-commands-as-they-are-executed) but while it makes the output more useful, it makes the script pretty unreadable.  What do you think?

Reply:
Btw, another problem is that (if selected and not installed) llvm gets installed by brew, which also installs Z3 using brew.  Maybe we need a version check for everything.

Reply:
In my dev environment, running this script is pretty rare, so noise while it runs isn't a problem, and if something goes wrong having the `-x` verbosity is useful to figure out what just happened.

Reply:
There is already a -v flag to the script to set -x.  Is that not enough for you?

Reply:
Ok, I've modified it to set VERBOSE if the shell is not interactive, which was the behavior described in https://github.com/aptos-labs/aptos-core/pull/8252:

"Also set -x in dev_setup.sh when running in CI only. Don't set it when running on laptops since it will become too verbose."